File: PdfRendererBasic/Application/src/main/java/com/example/android/pdfrendererbasic/PdfPagerAdapter.java
package com.example.android.pdfrendererbasic;

import android.graphics.Bitmap;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.viewpager.widget.PagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class PdfPagerAdapter extends PagerAdapter {

    private List<Bitmap> mBitmapList = new ArrayList<>();
    private ImageView mImageView;

    public PdfPagerAdapter(List<Bitmap> bitmap){
        this.mBitmapList = bitmap;
    }

    @Override
    public View instantiateItem(ViewGroup container, int position) {
        LayoutInflater inflater = LayoutInflater.from(container.getContext());
        View view  = inflater.inflate(R.layout.adapter_pager_pdf, container, false);
        mImageView = view.findViewById(R.id.image_pager_pdf);
        mImageView.setImageBitmap(mBitmapList.get(position));
        container.addView(view);
        return view;
    }

    @Override
    public int getCount() {
        return mBitmapList.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object){
        container.removeView((View) object);
    }
}

File: PdfRendererBasicKotlin/Application/src/main/java/com/example/android/pdfrendererbasic/PdfPagerAdapter.kt
package com.example.android.pdfrendererbasic

import android.graphics.Bitmap
import android.graphics.pdf.PdfRenderer
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.viewpager.widget.PagerAdapter
import java.io.File

class PdfPagerAdapter(val bitmap: List<Bitmap>): PagerAdapter() {

    override fun instantiateItem(container: ViewGroup, position: Int): View {
        val inflater = LayoutInflater.from(container.context)
        val view: View = inflater.inflate(R.layout.content_adapter, container, false)
        val image = view.findViewById<ImageView>(R.id.pdf_content)
        image.setImageBitmap(bitmap[position])
        container.addView(view)
        return view
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean = view == `object`

    override fun getCount(): Int = bitmap.size

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as View)
    }

}